### PR TITLE
Fix issue 1010

### PR DESCRIPTION
**Pull request by kroepke**
Add AMQP heartbeat timeout to input configuration.

### CODE REVIEW

File: graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java [hunk 1]
```diff
@@ -153,6 +155,7 @@ public void doLaunch(MessageInput input) throws MisfireException {
                 configuration.getInt(CK_PARALLEL_QUEUES),
                 configuration.getBoolean(CK_TLS),
                 configuration.getBoolean(CK_REQUEUE_INVALID_MESSAGES),
+                configuration.intIsSet(CK_HEARTBEAT_TIMEOUT) ? configuration.getInt(CK_HEARTBEAT_TIMEOUT) : ConnectionFactory.DEFAULT_HEARTBEAT,
```
Comment (joschi) [inline graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java]: We should ensure that the timeout is positive in this case.

Comment (kroepke) [inline graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java]: good point

File: graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java [hunk 2]
```diff
@@ -300,6 +311,16 @@ public ConfigurationRequest getRequestedConfiguration() {
             );
 
             cr.addField(
+                    new NumberField(
+                            CK_HEARTBEAT_TIMEOUT,
+                            "Heartbeat timeout",
+                            ConnectionFactory.DEFAULT_HEARTBEAT,
+                            "Heartbeat timeout interval in seconds (use 0 for no heartbeat)",
```
Comment (joschi) [inline graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java]: I'd rephrase that to "Heartbeat interval in seconds (use 0 to disable heartbeat)".

File: graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java [hunk 3]
```diff
@@ -140,6 +142,14 @@ public void setMessageAggregator(CodecAggregator aggregator) {
 
     @Override
     public void doLaunch(MessageInput input) throws MisfireException {
+        int heartbeatTimeout = ConnectionFactory.DEFAULT_HEARTBEAT;
+        if (configuration.intIsSet(CK_HEARTBEAT_TIMEOUT)) {
+            heartbeatTimeout = configuration.getInt(CK_HEARTBEAT_TIMEOUT);
+            if (heartbeatTimeout < 0) {
+                LOG.warn("AMQP heartbeat timeout {} should not be negative, disabling timeout.", heartbeatTimeout);
```
Comment (joschi) [inline graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java]: I'd rephrase that to:
```java
LOG.warn("AMQP heartbeat interval must not be negative, using default timeout ({}).", ConnectionFactory.DEFAULT_HEARTBEAT);
```

Comment (kroepke) [inline graylog2-inputs/src/main/java/org/graylog2/inputs/transports/AmqpTransport.java]: I added the configured timeout as well (for reference)

**Comment by joschi** [comment]
LGTM!

**Comment by kroepke** [comment]
thanks!
